### ISSUE TITLE
rewriteRules don't get submatches

### ISSUE THREAD

**Issue by fvsch** [BrowserSync/browser-sync#531]
Hi, and thanks for a very cool tool. :+1:

In my tests with 2.4.0 the fn method for a rewriteRule wouldn't get submatches.

```js
{
    rewriteRules: [
        {
            match: /\/(test1)\/(test2)\//,
            fn: function(found) {
                console.log(arguments)
                console.log(found.match(this.match))
            }
        }
    ],
}
```
Would it be possible to have the submatches directly as arguments (like when using a function in `str.replace(regexp, function(match, p1, p2){…})`)?

Also, it's obviously a bit late for that, but `match` is not a great name, since “a match” in English means what was actually found, not what we're looking for. So ideally we'd have something like:

```js
{
    rewriteRules: [
        {
            find: /regexp with submatches/,
            replaceWith: function(match, p1, p2) {…}
        }
    ],
}
```
But I digress. :)

**Comment by shakyShane** [comment]
Yeah I want this feature too. 

tbh, [resp-modifier](https://github.com/shakyShane/resp-modifier) (which both BrowserSync & Foxy use) is due a rewrite anyway, so if you want to send a PR with exactly what you need, then we can use that as a basis for the rewrite. See https://github.com/shakyShane/resp-modifier/blob/master/index.js#L76-L88

Re: the naming - it's `match:fn` to be consistent with resp-modifier which was originally a fork of [connect-livereload](https://github.com/intesso/connect-livereload), and although I totally agree with you, I'm more interested in consistency that of perfect naming :)

**Comment by shakyShane** [comment]
Fixed since 2.7.1
